skglm/estimators.py

```diff
@@ -914,12 +930,16 @@ def fit(self, X, y):
         self :
             Fitted estimator.
         """
+        if self.weights is None:
```

Comment:
Maybe I'm missing some context, but is there a reason for not checking the shape of weights to match the number of features?